appeals/web/src/server/app/components/file-downloader.component.js

```diff
@@ -198,3 +197,106 @@ export const getDocumentDownloadByVersion = async ({ apiClient, params, session
 
 	return response.status(200);
 };
+
+/**
+ *
+ * @param {import('@pins/blob-storage-client').BlobStorageClient} blobStorageClient
+ * @param {string} blobStorageContainer
+ * @param {string} blobStoragePath
+ * @returns {Promise<*>}
+ */
+export const createBlobDownloadStream = async (
+	blobStorageClient,
+	blobStorageContainer,
+	blobStoragePath
+) => {
+	// Document URIs are persisted with a prepended slash, but this slash is treated as part of the key by blob storage so we need to remove it
+	const documentKey = blobStoragePath.startsWith('/') ? blobStoragePath.slice(1) : blobStoragePath;
+
+	const blobProperties = await blobStorageClient.getBlobProperties(
+		blobStorageContainer,
+		documentKey
+	);
+
+	if (!blobProperties) {
+		// skip download
+		return;
+	}
+
+	const blobDownloadResponseParsed = await blobStorageClient.downloadStream(
+		blobStorageContainer,
+		documentKey
+	);
+
+	if (!blobDownloadResponseParsed?.readableStreamBody) {
+		throw new Error(`Document ${documentKey} missing stream body`);
+	}
+
+	// @ts-ignore
+	return blobDownloadResponseParsed.blobDownloadStream;
+};
+
+/**
+ * Build a zipped file and Download
+ *
+ * @param {{apiClient: import('got').Got, params: {caseId: string, filename?: string}, session: SessionWithAuth}} request
+ * @param {Response} response
+ * @returns {Promise<Response>}
+ */
+export const getBulkDocumentDownload = async ({ apiClient, params, session }, response) => {
+	const { filename: requestedFilename, caseId } = params;
+
+	// console.log(JSON.stringify({ getBulkDocumentDownload1: { caseId, requestedFilename } }, null, 2));
+
+	const folders = await getFolders(apiClient, caseId);
+
+	const bulkFileInfo = folders
+		?.filter((folder) => folder.documents.length)
+		.flatMap((folder) => {
+			return folder.documents.map((document) => {
+				const { blobStorageContainer, blobStoragePath, documentURI } =
+					document.latestDocumentVersion;
+				return {
+					fullName: `${folder.path}/${document.name}`,
+					blobStorageContainer,
+					blobStoragePath,
+					documentURI
+				};
+			});
+		});
+
```

Comment:
Here there should be a check for an empty `bulkFileInfo` (i.e. no files available).

Reply:
We should also have the ability to filter-out folders... e.g. internal and costs folders should probably be excluded (those will be the left part of the `folder.path` which usually indicates the area and the document type)